File: common/service_common/src/main/scala/o/lartifa/jij/model/JamConfig.scala
package o.lartifa.jij.model

import o.lartifa.jij.model.JamConfig.Biochronometer

/**
 * 果酱配置
 *
 * Author: sinar
 * 2021/4/10 21:32
 */
case class JamConfig
(
  name: String,
  qq: Long,
  masters: List[Long],
  masterCall: String = "主人",
  autoAcceptFriendRequest: Boolean = true,
  autoAcceptGroupRequest: Boolean = true,
  biochronometer: Option[Biochronometer] = None,
  behaviors: List[Behavior] = Nil
)

object JamConfig {
  case class Biochronometer(wakeUpAt: Short, sleepAt: Short, balderdash: List[String] = Nil)
}

File: common/service_common/src/main/kotlin/o/lartifa/jij/model/JamConfig.kt
package o.lartifa.jij.model

/**
 * 果酱配置
 *
 * Author: sinar
 * 2021/4/13 20:18
 */
data class JamConfig(
    val name: String,
    val qq: Long,
    val masterList: List<Long>,
    val masterCall: String = "主人",
    val autoAcceptFriendRequest: Boolean = true,
    val autoAcceptGroupRequest: Boolean = true,
    val biochronometer: Biochronometer?,
    val behaviors: List<Behavior> = emptyList()
) {
    companion object {
        data class Biochronometer(
            val wakeUpAt: Short,
            val sleepAt: Short,
            val balderdash: List<String> = emptyList()
        )
    }
}
